src/autopas/cells/FullParticleCell.h

```diff
@@ -90,6 +90,20 @@ class FullParticleCell : public ParticleCell<Particle> {
 
   std::array<double, 3> getCellLength() const override { return _cellLength; }
 
+  /**
+   * Returns the particle at position index. Needed by SingleCellIterator.
+   * @param index the position of the particle to return.
+   * @return the particle at position index.
+   */
+  decltype(auto) getParticle(size_t index) { return _particles.at(index); }
```

Comment:
I would prefer it if the function signature is written out, renamed to `at` and moved close to the `operator []` i.e.,
```suggestion
  Particle& at(size_t index) { return _particles.at(index); }
```
as this makes it easier to understand, especially for less experienced students.